firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/crcelements.txt

```diff
@@ -0,0 +1,212 @@
+      THEN("The output buffer is as expected")
+
+/* parse function
+Scenario: CRCElement: The parse method correctly updates its internal crc value and the constructor's payload buffer from the input buffer
+  
+  GIVEN("A CRCElement constructed with an empty payload buffer with a capacity of 254 bytes")
+    // assertions carried out in GIVEN:
+    // The crc accessor method returns 0
+    // The payload buffer returned by the payload accessor method is empty
+
+    WHEN("A body without a complete 4-byte header is parsed")
+      THEN("the parse method reports out of bounds status")
+      THEN("After the parse method is called, The value returned by the crc accessor method remains unchanged")
+      THEN("the constructor's payload buffer remains unchanged")
+
+    WHEN("A body with a complete 4-byte header, and a payload consistent with the CRC field of the header, is parsed")
+      THEN("the parse method reports ok status")
+      THEN("After the parse method is called, the value returned by the crc accessor method is equal to the crc field of the input_buffer body's header")
+      THEN("the payload returned from the payload accessor method is equal to the payload from the input_buffer body")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the CRCElement constructor") 
+      THEN("the payload given in the CRCElement constructor is independent of the input buffer body")
+
+    WHEN("A body with a complete 4-byte header, and a payload inconsistent with the CRC field of the header, is parsed")
+      THEN("the parse method reports ok status")
+      THEN("After the parse method is called, the value returned by the crc accessor method is equal to the crc field of the input_buffer body's header")
+      THEN("the value returned by the crc accessor method is inconsistent with the payload")
+      THEN("the payload returned from the payload accessor method is equal to the payload from the input_buffer body")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the CRCElement constructor")
+      THEN("the payload given in the CRCElement constructor is independent of the input buffer body")
+
+  GIVEN("A CRCElement constructed with a non-empty payload buffer of capacity 254 bytes")
+    // assertions carried out in GIVEN:
+    // The crc accessor method returns 0
+    // The payload buffer returned by the payload accessor method is '0x12 0x13 0x14 0x15 0x16'
+
+    WHEN("A body without a complete 4-byte header is parsed")
+      THEN("the parse method reports out of bounds status")
+      THEN("After the parse method is called, the value returned by the crc accessor method remains unchanged")
+      THEN("the payload accessor method returns a reference to the payload buffer given in the CRCElement constructor")
+      THEN("the payload given in the CRCElement constructor is independent of the input buffer body")      
```

Comment:
What should the contents of the payload buffer be after parse method is called? (I think the contents should be unchanged)